micrometer-java21/src/main/java/io/micrometer/java21/instrument/binder/jdk/VirtualThreadMetrics.java

```diff
@@ -90,12 +124,16 @@ public void close() {
         recordingStream.close();
     }
 
-    private record RecordingConfig(Duration maxAge, long maxSizeBytes, Duration pinnedThreshold) {
+    public void setActiveMetricEnabled(boolean activeMetricEnabled) {
+        this.activeMetricEnabled = activeMetricEnabled;
+    }
+
+    public record RecordingConfig(Duration maxAge, long maxSizeBytes, Duration pinnedThreshold) {
```

Comment:
If making this public, it should have a JavaDoc. We should probably also add some tests verifying usage we expect from users.